api/cwatch_api.php

```diff
@@ -287,65 +276,65 @@ public function getMalware($email, $domain)
      *     - path The path to the web directory for this site
      * @return CwatchResponse
      */
-    public function addScanner($email, $params)
+    public function addScanner($email, array $params)
     {
         $domainId = '';
-        $siteResponse = $this->getSite($email, $params['domain'], '', 'GET');
-        if (empty($siteResponse->errorMsg)) {
-            $site = json_decode($siteResponse->resp);
+        $siteResponse = $this->getSite($email, $params['domain']);
+        if (empty($siteResponse->errors())) {
+            $site = $siteResponse->response();
             $domainId = $site->engineSiteId;
         }
 
-        $response = $this->apiRequest('/domain/' . $domainId . '/settings/scanner/ftp', json_encode($params), 'POST');
-
-        return new CwatchResponse($response['content']);
+        return $this->apiRequest('/domain/' . $domainId . '/settings/scanner/ftp', $params, 'POST');
     }
 
     /**
      * Send an API request to cWatch server
      *
      * @param string $route The path to the API method
-     * @param string $body The data to be sent
+     * @param array $body The data to be sent
      * @param string $method Data transfer method (POST, GET, PUT, DELETE)
      * @return array
      */
-    private function apiRequest($route, $body, $method)
+    private function apiRequest($route, array $body, $method)
     {
         $url = $this->apiUrl . '/' . $route;
         $ch = curl_init();
 
-        curl_setopt($ch, CURLOPT_URL, $url);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-        curl_setopt($ch, CURLOPT_HEADER, 1);
-
         if ($method == 'POST') {
-            curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($body));
             curl_setopt($ch, CURLOPT_POST, 1);
         }
 
         if ($method == 'PUT') {
-            curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($body));
             curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'PUT');
         }
 
         if ($method == 'GET') {
+            $url .= empty($body) ? '' : '?' . urldecode(http_build_query($body));
```

Comment:
I think you can pass additional arguments to http_build_query so you don't need to call urldecode.

e.g.
`http_build_query($body, '', '&',  PHP_QUERY_RFC3986);`

Reply:
Hmm, that just changes the encoding rules, it doesn't stop the encoding.